### PR TITLE
Add lock for memory and standalone

### PR DESCRIPTION
**Pull request by soulasuna**
##  Add lock for memory and standalone

Related to #16269.

Changes proposed in this pull request:
-  Add lock for standalone
-  Add lock for memory

### CODE REVIEW

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    
+    private synchronized boolean innerTryLock(final long timeout) {
+        try {
+            if (innerLock.tryLock(timeout, TimeUnit.MILLISECONDS)) {
+                locked = true;
+            }
+            return false;
+        } catch (final InterruptedException isIgnored) {
+            return false;
+        }
```
Comment (sandynz) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java:56]: It's better to use `ignored`, but not `isIgnored`, keep the same style as before.

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+
+/**
+ * Non reentrant lock implemented ShardingSphereLock.
+ */
+@RequiredArgsConstructor
+public final class ShardingSphereNonReentrantLock implements ShardingSphereLock {
+
```
Comment (sandynz) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java:31]: Seems there're 2 the same `ShardingSphereNonReentrantLock`, could them be combined to one and shared for Memory and Standalone mode?

**Review by sandynz — CHANGES_REQUESTED** [review]

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java (hunk 2 above)
Comment (sandynz) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java:31]: After discussion, keep them for now, it'll be checked later.

**Review by sandynz — APPROVED** [review]

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java (hunk 1 above)
Comment (soulasuna) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java]: Fixed

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java:31]: Not yet, will be refactored after the basic functions are completed.

**Pull request merged**